File: day5/exercises-solutions/collective-one.c
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv)
{
    int mesg,buf,rank;

    MPI_Init(&argc, &argv);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    mesg = rank;
    MPI_Reduce(&mesg,&buf,1,MPI_INT,MPI_SUM,0,MPI_COMM_WORLD);
    if (rank == 0) printf("final result is: %d\n",buf);
    MPI_Finalize();
    return 0;
}

File: day5/exercises-solutions/collective-one.f90
PROGRAM collective_one
  USE mpi
  IMPLICIT NONE
  INTEGER :: rank, buf, mesg, ierr

  CALL MPI_INIT(ierr)
  CALL MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)

  mesg = rank
  CALL MPI_REDUCE(mesg, buf, 1, MPI_INTEGER, MPI_SUM, 0, MPI_COMM_WORLD, ierr)
  IF (rank == 0) PRINT*,"final result is:", buf

  CALL MPI_FINALIZE(ierr)
END PROGRAM collective_one
